api/models/series_test.go

```diff
@@ -90,3 +92,103 @@ func TestJsonMarshal(t *testing.T) {
 		}
 	}
 }
+
+func TestSetTags(t *testing.T) {
+	cases := []struct {
+		in  Series
+		out map[string]string
+	}{
+		{
+			in: Series{},
+			out: map[string]string{
+				"name": "",
+			},
+		},
+		{
+			in: Series{
+				Target: "a",
+			},
+			out: map[string]string{
+				"name": "a",
+			},
+		},
+		{
+			in: Series{
+				Target: `a\b`,
+			},
+			out: map[string]string{
+				"name": `a\b`,
+			},
+		},
+		{
+			in: Series{
+				Target: "a;b=c;c=d",
+			},
+			out: map[string]string{
+				"name": "a",
+				"b":    "c",
+				"c":    "d",
+			},
+		},
+		{
+			in: Series{
+				Target: "a;biglongtagkeyhere=andithasabiglongtagvaluetoo;c=d",
+			},
+			out: map[string]string{
+				"name":              "a",
+				"biglongtagkeyhere": "andithasabiglongtagvaluetoo",
+				"c":                 "d",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		c.in.SetTags()
+		if !reflect.DeepEqual(c.out, c.in.Tags) {
+			t.Fatalf("SetTags incorrect\nexpected:%v\ngot:     %v\n", c.out, c.in.Tags)
+		}
+	}
+}
+
+func BenchmarkSetTags_00tags_00chars(b *testing.B) {
+	benchmarkSetTags(b, 0, 0, 0)
+}
+
+func BenchmarkSetTags_20tags_32chars(b *testing.B) {
+	benchmarkSetTags(b, 20, 32, 32)
+}
+
+func benchmarkSetTags(b *testing.B, numTags, tagKeyLength, tagValueLength int) {
+	in := Series{
+		Target: "my.metric.name",
+	}
+
+	for i := 0; i < numTags; i++ {
+		in.Target = in.Target + ";" + randString(tagKeyLength)
+		in.Target = in.Target + "="
+		in.Target = in.Target + randString(tagValueLength)
```

Comment:
why not simplify this to
```
in.Target += ";" + randString(tagKeyLength) + "=" + randString(tagValueLength)
```
also, we could use https://golang.org/pkg/strings/#Builder here

Reply:
@shanson7 is it deliberate that you are still doing `in.Target = in.Target + ...` rather then `+=` ?

Reply:
Nope